sensor/kubernetes/listener/resources/networkpolicy_test.go

```diff
@@ -0,0 +1,445 @@
+						MatchLabels: map[string]string{
+							"app":  "sensor",
+							"role": "backend",
+						},
+					},
+				},
+			},
+			oldNetpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			action: central.ResourceAction_UPDATE_RESOURCE,
+			expectedSelector: []map[string]string{
+				{
+					"app":  "sensor",
+					"role": "backend",
+				},
+			},
+			expectedEmpty: false,
+		},
+		{
+			name: "Update NetworkPolicy, delete selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			oldNetpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+				Spec: &storage.NetworkPolicySpec{
+					PodSelector: &storage.LabelSelector{
+						MatchLabels: map[string]string{
+							"app":  "sensor",
+							"role": "backend",
+						},
+					},
+				},
+			},
+			action: central.ResourceAction_UPDATE_RESOURCE,
+			expectedSelector: []map[string]string{
+				{
+					"app":  "sensor",
+					"role": "backend",
+				},
+			},
+			expectedEmpty: false,
+		},
+		{
+			name: "Delete NetworkPolicy",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+				Spec: &storage.NetworkPolicySpec{
+					PodSelector: &storage.LabelSelector{
+						MatchLabels: map[string]string{
+							"app":  "sensor",
+							"role": "backend",
+						},
+					},
+				},
+			},
+			oldNetpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+				Spec: &storage.NetworkPolicySpec{
+					PodSelector: &storage.LabelSelector{
+						MatchLabels: map[string]string{
+							"app":  "sensor",
+							"role": "backend",
+						},
+					},
+				},
+			},
+			action: central.ResourceAction_REMOVE_RESOURCE,
+			expectedSelector: []map[string]string{
+				{
+					"app":  "sensor",
+					"role": "backend",
+				},
+			},
+			expectedEmpty: false,
+		},
+		{
+			name: "Delete NetworkPolicy, no selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			oldNetpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			action:           central.ResourceAction_REMOVE_RESOURCE,
+			expectedSelector: []map[string]string{},
+			expectedEmpty:    true,
+		},
+	}
+	for _, c := range cases {
+		if c.oldNetpol != nil {
+			nps.Upsert(c.oldNetpol)
+		}
+		sel, isEmpty := dispatcher.getSelector(c.netpol, c.action)
+		assert.Equal(t, isEmpty, c.expectedEmpty)
+		for _, s := range c.expectedSelector {
+			assert.True(t, sel.Matches(labels.Set(s)))
+		}
+
+		if c.oldNetpol != nil {
+			nps.Delete(c.oldNetpol.GetId(), c.oldNetpol.GetNamespace())
+		}
+	}
+}
+
+func TestUpdateDeploymentsFromStore(t *testing.T) {
+	if !features.NetworkPolicySystemPolicy.Enabled() {
+		return
+	}
+
+	mockCtrl := gomock.NewController(t)
+	nps := newNetworkPoliciesStore()
+	ds := newDeploymentStore()
+	det := mocks.NewMockDetector(mockCtrl)
+
+	dispatcher := newNetworkPolicyDispatcher(nps, ds, det)
```

Comment:
All these things can be grouped up in a `SetupTest()` if you use testify `suite.Suite`.

Reply:
Done